### PR TITLE
OU-632: Split ALERTS query_range into several requests for Incidents

### PR DESCRIPTION
**Pull request by rioloc**
This PR introduces the following changes:

## ALERTS expression simplified
- The previous expression used as _query_ param in the _query_range_ endpoint is simplified as
`ALERTS{}` . The second half of the query,  `+ on () group_left (component, layer) (absent(meta{layer="${query.layer}", component="${query.component}"})))`, was removed because  `component` and `layer` labels  are already present in the incident entity.

## Split of  `query_range` request into multiple ones for long queries
- When the query expression, composed by several `ALERTS{}` joined via `or` condition, exceeds a `MAX_URL_LENGTH` limit, then several queries will be created instead. 
- The `fetchDataForIncidentsAndAlerts` was updated to support both single and multiple queries. In the case of multiple queries then multiple requests will be triggered to the Prometheus API.

## skip alert if no incident is matched
- Added logic to skip the alert if no `matchinIncident` is found. This should be an edge case that could happen if an alert is triggered and a new incident metric was not produced yet.

Assisted-By: Claude Code

### CODE REVIEW

File: web/src/components/Incidents/processAlerts.ts [hunk 1]
```diff
@@ -265,16 +266,16 @@ export function convertToAlerts(
...
+        layer: matchingIncident.layer,
         name: alert.metric.name,
         alertstate: resolved ? 'resolved' : 'firing',
         values: paddedValues,
         alertsStartFiring: firstTimestamp,
         alertsEndFiring: lastTimestamp,
         resolved,
         x: 0, // Will be set after sorting
-        silenced,
+        silenced: matchingIncident.silenced,
```
Comment (falox) [inline web/src/components/Incidents/processAlerts.ts]: ok to assume the incident is not null but still needs ?? false, in case the old metric doesn't contain the silenced label. edge case, but better to be safe

**Comment by rioloc** [comment]
> make sure that the splitting + merging doesn't affect the ordering of the data points (probably covered in the unit tests, but I comment anyway).

Also for this the order should be guaranteed to be deterministic, so the one of the results of `customQuery | string[]`. 
In https://github.com/openshift/monitoring-plugin/pull/678/commits/6a724e5493d835a12b93e32f066afa6338afd8c4 I've added another unit to check result ordering is correct

File: web/src/components/Incidents/api.ts [hunk 2]
```diff
@@ -38,63 +61,91 @@ import { PROMETHEUS_QUERY_INTERVAL_SECONDS } from './utils';
...
+          resultType: 'matrix',
+          result: [],
+        },
+      } as PrometheusResponse);
+    }
+
+    return fetch(url);
+  });
+
+  const responses = await Promise.all(promises);
```
[the remark this answers is not in the document]
Comment (rioloc) [reply web/src/components/Incidents/api.ts]: I believe we want to be sure that in the case any of the subqueries fail then the an error is returned, otherwise the risk is to show partial informations to the user

File: web/src/components/Incidents/api.ts [hunk 3]
```diff
@@ -38,63 +61,91 @@ import { PROMETHEUS_QUERY_INTERVAL_SECONDS } from './utils';
...
+      return Promise.resolve({
+        status: 'success',
+        data: {
+          resultType: 'matrix',
+          result: [],
+        },
+      } as PrometheusResponse);
+    }
+
+    return fetch(url);
```
[the remark this answers is not in the document]
Comment (rioloc) [reply web/src/components/Incidents/api.ts]: Fixed in https://github.com/openshift/monitoring-plugin/pull/678/commits/6dc32da035aec7e4cfb967c56a5e1d7f292914b7

Do you believe is there a better way than the used one to mock `consoleFetchJSON` in unit tests?
